app/include/zmk/behavior.h

```diff
@@ -26,6 +26,7 @@ struct zmk_behavior_binding_event {
     int layer;
     uint32_t position;
     int64_t timestamp;
+    uint8_t source;
```

Comment:
I do wonder of the addition for the `source` in a few places should be conditional on `CONFIG_ZMK_SPLIT`. On more resource limited devices, we may be storing things like the queue items, etc. in RAM, and the extra 4-bytes will be not-insignificant, especially if we're buffering 32 of them, etc.

Reply:
Valid concern, let me see if I can `#ifdef` guard these and whether it will be very ugly.

Reply:
I refactored all new `source` uses to be guarded by the split symbol, tested on split and non-splits. 

In some cases I thought it was nicer to pass a reference to the `zmk_behavior_binding_event` struct rather than the `position` (+`state` for split). Can you have another look?